### PR TITLE
lookupObject at most one updated

### PR DESCRIPTION
**Pull request by olegosh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,3 +30,28 @@ Executes custom request
...
 * **HTTP headers** - (object, required) HTTP headers of the response, required.
 * **Response Body** - (object, optional) HTTP response body.
+
+### Lookup Object (at most 1) by Unique Criteria
+
+Lookup a single object by a selected field that uniquely identifies it in the database.
+
+#### Configuration Fields
+
+- Object Type (required): One of the six object types available in the database.
```
Comment (EmptyInfinity) [inline README.md:40]: add ```(datatype, required/optional)``` for each field

File: component.json [hunk 2]
```diff
@@ -41,6 +41,63 @@
         "in": "./src/schemas/rawRequest.in.json",
         "out": "./src/schemas/rawRequest.out.json"
       }
+    },
+    "lookupObject": {
+      "main": "./lib/actions/lookupObject.js",
+      "title": "Lookup Object by Unique Criteria",
```
Comment (EmptyInfinity) [inline component.json:47]:
```suggestion
      "title": "Lookup Object (at most 1) by Unique Criteria",
```

File: src/actions/lookupObject.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  };
+
+  if (cfg.allowIdToBeOmitted) {
+    meta.in.properties.lookupCriteriaValue.required = false;
+  }
+
+  return meta;
+};
+
+exports.getLookupCriteria = async function (cfg: { objectType: any; }) {
```
Comment (EmptyInfinity) [inline src/actions/lookupObject.ts:78]:
```suggestion
exports.getLookupCriteria = async function (cfg: { objectType: string; }) {
```

File: src/client.ts [hunk 4]
```diff
@@ -14,4 +14,61 @@ export default class Client {
   async apiRequest(opts: AxiosRequestConfig) {
     return axios.request(opts);
   }
+
```
Comment (EmptyInfinity) [inline src/client.ts:17]: remove comments pls

File: src/actions/lookupObject.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
+import Client from '../client';
+
+const { messages } = require('elasticio-node');
+
+// eslint-disable-next-line max-len
+exports.process = async function ProcessAction(msg: { body: { lookupCriteriaValue: any; }; }, cfg: { allowIdToBeOmitted?: any; baseUrl?: any; allowZeroResults?: any; objectType?: any; lookupCriteria?: any; }) {
```
Comment (EmptyInfinity) [inline src/actions/lookupObject.ts:6]: You may use `async function ProcessAction(msg: any, cfg: any)`
And if you want to define parameters structure - pls use it as variable

File: README.md (hunk 1 above)
Comment (olegosh) [reply README.md]: done

File: README.md [hunk 6]
```diff
@@ -5,6 +5,7 @@
 * [Description](#description)
 * [Actions](#actions)
    * [Make Raw Request](#make-raw-request)
+   * [Lookup Object (at most 1) by Unique Criteria](#lookup-object-at-most-1-by-unique-criteria)
```
[the remark this answers is not in the document]
Comment (olegosh) [reply README.md]: done

File: component.json [hunk 7]
```diff
@@ -41,6 +41,63 @@
...
+          "viewClass": "SelectView",
+          "required": true,
+          "order": 5,
+          "model": "getUniqueFieldsModel"
+        },
+        "allowCriteriaToBeOmitted": {
+          "label": "Allow criteria to be omitted",
+          "viewClass": "CheckBoxView",
+          "order": 4,
+          "note": "If selected, the lookup criteria field is not required"
```
[the remark this answers is not in the document]
Comment (olegosh) [reply component.json]: done

File: package.json [hunk 8]
```diff
@@ -26,6 +26,7 @@
   },
   "dependencies": {
     "axios": "0.24.0",
+    "chai-as-promised": "^7.1.1",
```
[the remark this answers is not in the document]
Comment (olegosh) [reply package.json]: done

File: src/actions/lookupObject.ts (hunk 3 above)
Comment (olegosh) [reply src/actions/lookupObject.ts]: done

File: src/actions/lookupObject.ts (hunk 5 above)
Comment (olegosh) [reply src/actions/lookupObject.ts]: done

**Review by EmptyInfinity — APPROVED** [review]

**Pull request merged**